app/controllers/expense_transactions_controller.rb

```diff
@@ -0,0 +1,50 @@
+class ExpenseTransactionsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :create_new_form, only: %i[new create]
+  before_action :set_categories, only: %i[new create]
```

Comment:
@marderer why not just:
```ruby
before_action :set_categories

def create
  if @form.validate(permitted_params)
    ExpenseTransactions::Creator.new(create_params).create
    redirect_to activity_page_path, flash: {notice: t('transaction_create')}
  else
    render :new
  end
end

def form
  @_form ||= ExpenseTransactionForm.new(
    current_user.transactions.new,
    expense_transactions: ExpenseTransaction.new
  )
end

```

Reply:
@AleksSenkou I'm using `@form` for checks validation before creating